ext/RMagick/rmimage.c

```diff
@@ -11646,6 +11654,10 @@ rd_image(VALUE class ATTRIBUTE_UNUSED, VALUE file, reader_t reader)
     Image *images;
     ExceptionInfo *exception;
 
+#if !defined(_WIN32)
+    signal(SIGCHLD, sig_handler);
```

Comment:
ImageMagick API will read PDF file using `fork/waitpid`.
When child process will be finished, Ruby process will raise signal `SIGCHLD`
And if uncaught the signal, waitpid() will return error and ImageMagick will raise exception.

To ignore `SIGCHLD` raised by child process, it need `signal()`.